### PR TITLE
test: add unit tests

### PR DESCRIPTION
**Pull request by vegerot**
- Added tests from README
- Refactor a bit of the code to make it more easily testable

### CODE REVIEW

**Comment by vegerot** [comment]
It sounds like I should add a test for UTF-8 compatibility

File: util.go [hunk 1]
```diff
@@ -115,3 +115,12 @@ func prettier(expr metricsql.Expr, ident int) []byte {
 
 	return buf
 }
+
+func FormatString(s string) string {
```
Comment (jiacai2050) [inline util.go]: - How about name this function `Prettier`

File: util.go [hunk 2]
```diff
@@ -115,3 +115,12 @@ func prettier(expr metricsql.Expr, ident int) []byte {
...
+	expr, err := metricsql.Parse(s)
+	if err != nil {
+		panic(err)
```
Comment (jiacai2050) [inline util.go]: it should return err out

File: main_test.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
+package main_test
```
Comment (jiacai2050) [inline main_test.go:1]: I think we can just put this file in `main` package

File: main_test.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+import (
+	"testing"
+
+	main "promql-prettier"
+)
+
+func TestFormatting(t *testing.T) {
+	type gotWant struct {
```
Comment (jiacai2050) [inline main_test.go]:
```
tests := []struct {
input string
expected string
}{
  {},
  {},
..
}
```

**Review by jiacai2050 — CHANGES_REQUESTED** [review]

File: util.go (hunk 1 above)
Comment (vegerot) [reply util.go]: Good idea

File: util.go (hunk 2 above)
Comment (vegerot) [reply util.go]: thanks

File: main_test.go (hunk 4 above)
Comment (vegerot) [reply main_test.go]: I like that!

File: main_test.go (hunk 3 above)
Comment (vegerot) [reply main_test.go:1]: I prefer putting my tests in a different package to make sure I'm only testing exported fields

For `main` it doesn't really matter either way, but I think this is more idiomatic

Comment (vegerot) [reply main_test.go:1]: See https://pkg.go.dev/testing and https://github.com/golang/go/issues/25223#issuecomment-1201120914

Comment (vegerot) [reply main_test.go:1]: If you feel strongly about the package naming I will change it

Comment (jiacai2050) [reply main_test.go:1]: Good point

**Review by jiacai2050 — APPROVED** [review]
Thanks

**Pull request merged**